### PR TITLE
Julian 004

### PR DESCRIPTION
**Pull request by jlbprof**
This pushes an xml (xbel) format bookmark file to the server.

I had to html encode the bookmark href's that may be a problem, but if I did not I could not get the xbel to validate with complex url's (such as some with question marks and ampersands).

### CODE REVIEW

File: src/lib/Account.js [hunk 1]
```diff
@@ -125,6 +125,9 @@ export default class Account {
 
   async sync () {
     try {
+      if ('syncStart' in this.server)
+        await this.server.syncStart ();
+
       if (this.getData().syncing) return
       console.log('Starting sync process for account ' + this.getLabel())
       await this.setData({...this.getData(), syncing: true})
```
Comment (marcelklehr) [inline src/lib/Account.js]: `syncStart` should be triggered after this block, to make sure that no two tasks are syncing the same account ;)

File: src/lib/Account.js [hunk 2]
```diff
@@ -161,6 +164,10 @@ export default class Account {
       await this.tree.removeOrphanedFolders()
 
       await this.setData({...this.getData(), error: null, syncing: false, lastSync: Date.now()})
+
+      if ('syncComplete' in this.server)
+        await this.server.syncComplete ();
+
```
Comment (marcelklehr) [inline src/lib/Account.js]: should be before the call to setData() :)

File: src/lib/adapters/WebDav.js [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+            rStatus = await this.checkLock ();
+			console.log ("obtainLock: 002 " + rStatus);
+            if (rStatus == 200) {
+				console.log ("waiting timeout :" + increment + ":");
+				await this.timeout (increment * 1000);
+				console.log ("waited timeout :" + increment + ":");
+            } else if (rStatus == 404) {
+				break;
+            }
+        }
```
Comment (marcelklehr) [inline src/lib/adapters/WebDav.js]: nice use of await! :)

File: src/lib/adapters/WebDav.js [hunk 4]
```diff
@@ -0,0 +1,593 @@
...
+
+        if (response.status !== 200) {
+            return {
+                'status' : response.status,
+                'db' : new Map ()
+            };
+        }
+
+        let bookmark_array = await response.json();
```
Comment (marcelklehr) [inline src/lib/adapters/WebDav.js]: You'll probably want to parse XBEL here, too :D

**Comment by marcelklehr** [comment]
Looks good so far. Two things: Could you filter out my commits from the master branch? Also, the indentation is inconsistent between tabs and spaces. I can fix that after pulling, as well, though.

**Comment by jlbprof** [comment]
which do you prefer, tabs or spaces?  I will convert.

Regarding the filtering of commits, not sure what you mean.

**Comment by marcelklehr** [comment]
> which do you prefer, tabs or spaces?  I will convert.

Two spaces would be nice :)

> Regarding the filtering of commits, not sure what you mean.

In the commits above you can see some commits I made on the master branch, e.g. the v2.2.4 release and some others. Those conflict with the master branch and prevent this branch from being able to be pulled.

**Comment by marcelklehr** [comment]
You could `git rebase -i` and simply drop them.

**Comment by jlbprof** [comment]
I removed the v2.2.5 and v2.2.4 commits and there are no conflicts right now.

**Pull request closed without merging**